Net/HttpRequestBuilder.cs

```diff
@@ -80,71 +105,102 @@ public HttpRequestBuilder SetHeader(string key, string value)
     {
-        return _response ?? throw new InvalidOperationException("在请求完成前的意外调用");
+        _request.Version = version;
+        return this;
     }
-    /// <summary>
-    /// 读取响应载荷
-    /// </summary>
-    /// <returns>string</returns>
-    public string ReadResponseAsString()
+
+    public HttpRequestBuilder WithLoggingOptions(bool doLog)
     {
-        return ReadResponseAsStringAsync().GetAwaiter().GetResult();
+        _doLog = doLog;
+        return this;
     }
-    /// <summary>
-    /// 读取响应载荷 （异步）
-    /// </summary>
-    /// <returns>string</returns>
-    public async Task<string> ReadResponseAsStringAsync()
+
+    public HttpRequestBuilder WithCompletionOption(HttpCompletionOption option)
     {
-        return await GetResponse().Content.ReadAsStringAsync();
+        _completionOption = option;
+        return this;
     }
+
     /// <summary>
-    /// 读取响应载荷
+    ///
     /// </summary>
-    /// <returns>byte[]</returns>
-    public byte[] ReadResponseAsByteArray()
+    /// <param name="throwIfNotSuccess">请求失败时是否抛出异常</param>
+    /// <param name="retryTimes">请求重试次数</param>
+    /// <param name="retryPolicy">依据请求当前尝试的次数给出的重试时长控制方法</param>
+    /// <exception cref="HttpRequestException">要求 <paramref name="throwIfNotSuccess"/> 时并且 HTTP 请求失败</exception>
+    /// <returns></returns>
+    public async Task<HttpResponseHandler> SendAsync(bool throwIfNotSuccess = false, int retryTimes = 3, Func<int,TimeSpan>? retryPolicy = null)
     {
-        return ReadResponseAsByteArrayAsync().GetAwaiter().GetResult();
-    }
-    /// <summary>
-    /// 读取响应载荷（异步）
-    /// </summary>
-    /// <returns>byte[]</returns>
-    public async Task<byte[]> ReadResponseAsByteArrayAsync()
-    { 
-        return await GetResponse().Content.ReadAsByteArrayAsync();
+        // 处理 Cookies
+        if (_cookies.Count != 0)
+        {
+            if (_request.Headers.Contains("Cookie")) _request.Headers.Remove("Cookie"); //去掉野生的饼干
+            var cookiesCtx = new StringBuilder(_cookies.Count * 30); //后期需要根据实际使用调整预分配的容量大小以提高文本构建性能
+            foreach (var cookie in _cookies)
+            {
+                if (cookiesCtx.Length > 0)
+                    cookiesCtx.Append("; ");
+                cookiesCtx
+                    .Append(Uri.EscapeDataString(cookie.Key))
+                    .Append('=')
+                    .Append(_getSafeCookieValue(cookie.Value));
+            }
+            _request.Headers.TryAddWithoutValidation("Cookie", cookiesCtx.ToString());
+        }
+
+        var client = NetworkService.GetClient();
+        _makeLog($"向 {_request.RequestUri} 发起 {_request.Method} 请求");
+        var responseMessage = await NetworkService.GetRetryPolicy(retryTimes, retryPolicy)
+            .ExecuteAsync(async () => await client.SendAsync(_request, _completionOption));
+        if (responseMessage.RequestMessage?.RequestUri is not null &&
+            !_request.RequestUri.Equals(responseMessage.RequestMessage.RequestUri))

```

Comment:
Why not use operator?
```suggestion
        if (responseMessage.RequestMessage?.RequestUri != null &&
            _request.RequestUri != responseMessage.RequestMessage.RequestUri)
```